test/remote/gateways/remote_reach_test.rb

```diff
@@ -74,6 +75,60 @@ def test_failed_capture
     assert_equal 'Not Found', response.message
   end
 
+  #
+  # The following refund test was commented or it is optional because take
+  # ~1 minute to get a successful response
+  #
+
+  # def test_successful_refund
+  #   purchase = @gateway.purchase(@amount, @credit_card, @options)
+  #   sleep(1.minute)
+  #   response = @gateway.refund(@amount, purchase.authorization, { reference_id: 'REFUND_TAG' })
+
+  #   assert_success response
+  #   assert response.params['response'][:RefundId].present?
+  # end
+
+  def test_failed_refund
+    purchase = @gateway.purchase(@amount, @credit_card, @options)
+    response = @gateway.refund(@amount, purchase.authorization, { reference_id: 'REFUND_TAG' })
+
+    assert_failure response
+    assert_equal 'OrderStateInvalid', response.params['response'][:Error][:Code]
+    assert response.params['response'][:OrderId].present?
+  end
+
+  def test_successful_void
+    authorize = @gateway.authorize(@amount, @credit_card, @options)
+    response = @gateway.void(authorize.authorization, @options)
+
+    assert_success response
+    assert response.params['response'][:OrderId].present?
+  end
+
+  def test_failed_void
+    response = @gateway.void(@non_valid_authorization, @options)
+    assert_failure response
+
+    assert_equal 'Not Found', response.message
+    assert response.params.blank?
+  end
+
+  def test_successful_verify
+    response = @gateway.verify(@credit_card, @options)
+
+    assert_success response
+    assert response.params['response'][:OrderId].present?
+  end
+
+  def test_failed_verify
+    response = @gateway.verify(@declined_card, @options)
+
+    assert_failure response
+    assert response.params['response'][:OrderId].present?
+    assert_equal 'PaymentAuthorizationFailed', response.params['response'][:Error][:Code]
```

Comment:
💬  Comment:

Same comment here about the error_code.

Reply:
ok, done!